funsor/pyro/hmm.py

```diff
@@ -207,8 +197,8 @@ def log_prob(self, value):
         result = self._trans + obs
         result = sequential_sum_product(ops.logaddexp, ops.add,
                                         result, "time", {"state": "state(time=1)"})
-        result = self._init + result.reduce(ops.logaddexp, "state(time=1)")
-        result = result.reduce(ops.logaddexp, "state")
+        result += self._init
```

Comment:
@martinjankowiak @fehiepsi This reorders so that `.reduce()` is called only after adding in the initial state prior, which improves conditioning of Cholesky decompositions inside `.reduce()`. This reordering fixed lots of actual errors in the `SwitchingLinearHMM`, and in the `GaussianHMM` allowed me to reduce test tolerance from 1e-4 to 1e-5.